src/fs/serve_dir.rs

```diff
@@ -70,7 +71,7 @@ impl<State> Endpoint<State> for ServeDir {
             res.set_body(body);
 
             if let Some(content_type) = mime_guess::from_path(&file_path).first() {
-                res = res.set_mime(content_type);
+                res = res.set_mime(content_type.to_string().parse::<Mime>().unwrap());
```

Comment:
This is not very nice, should `http_types::HeaderValue` have an into for this?

Reply:
I think this code is temporary and will be replaced by the mime sniffing stuff in http_types

Reply:
This is converting from a mime::Mime to a http_types::Mime, and I think the goal is for neither tide nor http_types to have a dep on the mime crate

Reply:
http-types Body now has a `from_file` constructor that applies the inference. Could as maybe use that here?

The trickiest part I see is checking the error type so that we can set the right status codes, but that should be possible through downcasting.

That way we can get rid of the `mime` dep effective immediately!

Reply:
Addressed in 5000e1c